kernel/src/madt.c

```diff
@@ -0,0 +1,103 @@
+#include "madt.h"
+
+#include "acpi.h"
+#include "kassert.h"
+#include "memory/paging.h"
+#include "io_apic.h"
+
+// Conversion table from 8259 IRQs to GSI
+GSIOverrideEntry g_gsi_override_table[MAX_8259_IRQ_COUNT];
+
+// References to the MADT's IOAPIC entries, uses g_ioapic_count
+IOAPICInfo g_found_ioapics[MAX_IOAPIC_COUNT];
+uint64_t g_ioapic_count = 0;
+
+// References to the MADT's Local APIC entries, uses g_lapic_count
+LAPICInfo* g_found_lapics[MAX_LAPIC_COUNT];
+uint64_t g_lapic_count = 0;
+
+// Local APIC's physical address
+PhysicalAddress g_lapic_physical_address = 0xFEE00000UL;
```

Comment:
You shouldn't hardcode the local APIC address, it's provided in MADT header: https://wiki.osdev.org/MADT#Table_Structure